pkg/data/account.go

```diff
@@ -0,0 +1,172 @@
+			ConditionExpression:       expr.Condition(),
+			ExpressionAttributeNames:  expr.Names(),
+			ExpressionAttributeValues: expr.Values(),
+			// Return the updated record
+			ReturnValues: aws.String(returnValue),
+		},
+	)
+	var awsErr awserr.Error
+	if gErrors.As(err, &awsErr) {
+		if awsErr.Code() == "ConditionalCheckFailedException" {
+			return fmt.Errorf(
+				"unable to update account %v no account exists with "+
+					"LastModifiedOn=\"%v\": %w",
+				account.ID, *account.LastModifiedOn, errors.ErrConflict)
+		}
+	}
+	if err != nil {
+		return fmt.Errorf("update failed for account %s: %s: %w", *account.ID, err, errors.ErrInternalServer)
+	}
+
+	return nil
+}
+
+// Delete the Account record in DynamoDB
+func (a *Account) Delete(account *model.Account) error {
+
+	res, err := a.AwsDynamoDB.DeleteItem(
+		&dynamodb.DeleteItemInput{
+			// Query in Lease Table
+			TableName: aws.String(a.TableName),
+			// Return the updated record
+			ReturnValues: aws.String("ALL_NEW"),
+			Key: map[string]*dynamodb.AttributeValue{
+				"Id": {
+					S: account.ID,
+				},
+			},
+		},
+	)
+
+	if err != nil {
+		return fmt.Errorf("failed to delete account %s: %s: %w", *account.ID, err, errors.ErrInternalServer)
+	}
+
+	return dynamodbattribute.UnmarshalMap(res.Attributes, &account)
+}
+
+// GetAccountByID the Account record by ID
+func (a *Account) GetAccountByID(accountID string, account *model.Account) error {
+
+	res, err := a.AwsDynamoDB.GetItem(
+		&dynamodb.GetItemInput{
+			// Query in Lease Table
+			TableName: aws.String(a.TableName),
+			Key: map[string]*dynamodb.AttributeValue{
+				"Id": {
+					S: aws.String(accountID),
+				},
+			},
+			ConsistentRead: aws.Bool(a.ConsistentRead),
+		},
+	)
+
+	if err != nil {
+		return fmt.Errorf("failed to get account %s: %s: %w", accountID, err, errors.ErrInternalServer)
+	}
+
+	if len(res.Item) == 0 {
+		return fmt.Errorf("account %s not found: %w", accountID, errors.ErrNotFound)
+	}
+
+	return dynamodbattribute.UnmarshalMap(res.Item, &account)
+}
+
+// GetAccountsByStatus - Returns the accounts by status
+func (a *Account) GetAccountsByStatus(status string) (*model.Accounts, error) {
+
+	res, err := a.AwsDynamoDB.Query(&dynamodb.QueryInput{
+		TableName: aws.String(a.TableName),
+		IndexName: aws.String("AccountStatus"),
+		ExpressionAttributeValues: map[string]*dynamodb.AttributeValue{
+			":status": {
+				S: aws.String(string(status)),
+			},
+		},
+		KeyConditionExpression: aws.String("AccountStatus = :status"),
+		ConsistentRead:         aws.Bool(a.ConsistentRead),
+	})
+	if err != nil {
+		return nil, fmt.Errorf("error getting accounts by status %s: %s: %w", status, err, errors.ErrInternalServer)
+	}
+
+	accounts := &model.Accounts{}
+	err = dynamodbattribute.UnmarshalListOfMaps(res.Items, accounts)
+	return accounts, err
+}
+
+// GetAccounts Get a list of accounts based on Principal ID
+func (a *Account) GetAccounts(q *model.Account) (*model.Accounts, error) {
+	var expr expression.Expression
```

Comment:
Let's talk about pagination.
Maybe don't need to get that question answered in this PR, but i don't want to lose track of it

Reply:
I have an idea for that too.  I'm not sure it can be handled as cleanly but I think there is a solution that will work and put into a similar model as everything else.